MetaMask/metamask-extension
# fix: Delete migration 115

## **Description**

https://github.com/MetaMask/metamask-extension/pull/24369 reverted the commit that introduced migration 115. Instead of deleting 115, it modified it. The problem with this is:
- the migration now does nothing
- the migration can't just be updated in the future, because after it runs once for a user, it will never be run again

What we need to do is delete migration 115 and then rename the subsequent migrations to one number lower (and update their contents to reflect these new migration numbers. Then, when the reverted changes are restored, the migration can be re-added as the new highest numbered migration.

This PR makes the changes needed to delete migration 115 over multiple commits, for ease of review. Reviewing this PR is really easy _if_ you review commit by commit (but Github can't easily display the simplicity of the changes when they are all together). Below I have briefly described each commit and included a screenshot of what you will see if you click the link to look at that commit directly

e7d4cb55230f21782562e0c712e16b513cc8de29 **Deletes** the existing migration 115 files
![Screenshot from 2024-05-21 05-03-03](https://github.com/MetaMask/metamask-extension/assets/7499938/f062276d-5fa8-4857-880a-4ae1daefb229)


8d78df3060a4758fd5ea3b61de45863b44315849 **Renames** the existing migration 116 files to migration 115 files
![Screenshot from 2024-05-21 05-03-23](https://github.com/MetaMask/metamask-extension/assets/7499938/199e8d94-5101-472a-83b7-13b2d94037b9)


3023d17bb80d8d1137f45ff9ef0d1c3d2e33495b **Renames** the existing migration 117 files to migration 116 files
![Screenshot from 2024-05-21 05-03-39](https://github.com/MetaMask/metamask-extension/assets/7499938/7fddc54d-8e0a-44f7-ad14-00cf12041f14)


23f121b53d5bd40c1944f515be9d4b2ce349f753 **Renames** the existing migration 118 files to migration 117 files
![Screenshot from 2024-05-21 05-04-15](https://github.com/MetaMask/metamask-extension/assets/7499938/7c97a3aa-7f22-4db2-a7b7-bacbd9fca7b9)


f6cbb92cae17b563fb97e565d511b3cb0da45a11 **Renames** the existing migration 119 files to migration 118 files
![Screenshot from 2024-05-21 05-04-28](https://github.com/MetaMask/metamask-extension/assets/7499938/223cc3f9-b3a2-4b42-ae9a-3eba526d8968)


f40afd721ee5d535cb7506b13b17d810c1983426 **Modifies** the renamed files from the previous commits, so that any number within those files in the range 115-119 is replaced by the number one below.
![Screenshot from 2024-05-21 05-05-29](https://github.com/MetaMask/metamask-extension/assets/7499938/3a680ca8-1c30-4e6e-bace-7c8b206cdb7e)

088ecd5e125e90d44f575714836ee7c68ca89df4 Just deletes 1 line from app/scripts/migrations/index.js


[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24666?quickstart=1)

## **Manual testing steps**

1. Create a local build on the master branch, install and onboard
2. Create a build on this branch, install
3. In the background console, you should see logs that show migrations 115 to 118 running successfully

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
